src/dataExplorer/context/fluxQueryBuilder.tsx

```diff
@@ -120,16 +145,28 @@ export const FluxQueryBuilderProvider: FC = ({children}) => {
       name: measurement,
     } as ExecuteCommandInjectMeasurement)
 
-    // Reset fields and tags
+    // Reset fields, tags, selected tag values
     resetFields()
     resetTags()
+    setSelectedTagValues(DEFAULT_SELECTED_TAG_VALUES)
 
-    // Get fields and tags
+    // Fetch fields and tags
     getFields(selection.bucket, measurement)
     getTagKeys(selection.bucket, measurement)
   }
 
   const handleSelectField = (field: string): void => {
+    let fields = []
+    if (selection.fields?.includes(field)) {
+      // remove the selected field
+      fields = selection.fields.filter(item => item !== field)
+    } else {
```

Comment:
yeah. Treating the checkbox onClick() as a state toggle (rather than getting a boolean value returned) annoys me. But the List component from clockface and the SelectorList is existing code with this interface. It is what it is. 🤷🏼‍♀️